app/src/main/java/me/saket/dank/ui/user/UserSessionRepository.java

```diff
@@ -22,20 +32,26 @@
 public class UserSessionRepository {
 
   private static final String KEY_LOGGED_IN_USERNAME = "logged_in_username_v0.6.1";
+  private static final String KEY_SORTING_AND_PERIOD = "sorting_and_period_";
   private static final String EMPTY = "";
 
   private Lazy<Reddit> reddit;
   private final Preference<String> loggedInUsername;
+  private final Preference<String> sortingAndPeriod;
+  private HashMap<String, String> sortHashMap;
 
   @Inject
   public UserSessionRepository(Lazy<Reddit> reddit, @Named("user_session") RxSharedPreferences rxSharedPreferences) {
     this.reddit = reddit;
     loggedInUsername = rxSharedPreferences.getString(KEY_LOGGED_IN_USERNAME, EMPTY);
+    sortingAndPeriod = rxSharedPreferences.getString(KEY_SORTING_AND_PERIOD, EMPTY);
```

Comment:
`SharedPreferences` are supposed to be used for tiny key value pairs. Storing a big HashMap like this one might not be a good idea. It'll grow over time. Especially if we want to start storing the last used sorting for comments as well.

SharedPreference files are read on the main thread during app startup. As the file grows, the penalty will also grow. I think we should think of creating a separate repository class that will handle retaining this data asynchronously.